File: jungu/Data structure/10845.cpp
#include <cstdio>
#include <string>
#include <queue>
#include <iostream>

using namespace std;

int main() {

	int n;
	queue <int> q;
	cin >> n;

	while (n--) {
		string order;
		cin >> order;
		if (order == "push") {
			int n;
			cin >> n;
			q.push(n);
		}
		else if (order == "pop") {
			if (q.size() != 0) {
				printf("%d \n", q.front());
				q.pop();
			}
			else printf("-1 \n");
		}
		else if (order == "size")
			printf("%d \n", q.size());
		else if (order == "empty")
			printf("%d \n", q.empty());
		else if (order == "front") {
			if (q.size() != 0)
				printf("%d \n", q.front());
			else printf("-1 \n");
		}
		else if (order == "back") {
			if (q.size() == 0)
				printf("-1 \n");
			else printf("%d \n", q.back());
		}
	}

}

File: yoh/lec02/10845.py
import sys
inin = lambda: sys.stdin.readline().strip()
queue = []

for i in range(int(inin())):
	command = inin().split()
	if command[0] == 'push':
		queue.append(int(command[1]))
	elif command[0] == 'pop':
		if queue == []:
			print -1
		else:
			print queue.pop(0)
	elif command[0] == 'size':
		print len(queue)
	elif command[0] == 'empty':
		if queue == []:
			print 1
		else:
			print 0
	elif command[0] == 'front':
		if queue == []:
			print -1
		else:
			print queue[0]
	elif command[0] == 'back':
		if queue == []:
			print -1
		else:
			print queue[-1]